freud/data.py

```diff
@@ -233,6 +233,56 @@ def hex(cls):
         fractions = np.array([[0, 0, 0], [0.5, 0.5, 0]])
         return cls([1, np.sqrt(3)], fractions)
 
+    @classmethod
+    def from_lattice_vectors(
```

Comment:
Since this method (and `from_box_lengths_and_angles`) just replaces two lines of user script, I don't think it's needed. I am open to discussion though.

Reply:
It's just a convenience function. I love these. If you strongly feel they are not needed we can remove them. Up to you.

Reply:
My personal logic is that if we have a convenience function for making `UnitCell`'s in the `data` module for `Box.from_lattice_vectors` and `Box.from_box_lengths_and_angles`, then we should also have a convenience function for every factory method in the `Box` class (like `cube`, `from_box`, `from_matrix`, `square`, etc.). There are quite a few of those and it may get cumbersome to have so many convenience functions hanging around.

If you like having that convenience function, add one to your python scripts. It only takes two lines to write it, after all :)

Reply:
While I see what you're saying, there are a significant number of users/use cases for this specific functionality: making the translation between freud boxes and data from crystallography files more transparent is a good thing in my opinion. I think adding this particular convenience function provides enough to users that we can probably do without the rest mentioned above. If this truly doesn't fit though, I can live without it.

Reply:
What data does a crystallography file typically contain? If its a common use case to make a freud `UnitCell` from crystallography file data, then it makes sense to have a convenience function for it. I personally don't work a lot with those, so I'm not entirely sure what would be most helpful for others who work with it.

I think this topic is worth further offline discussion as well.

Reply:
CIF files (the most common file type, at least for computational/experimental researchers) have unit cell information encoded as a/b/c/α/β/γ. For DFT peeps, VASP and ABINIT provide lattice vectors so those would work without this function.

[Crystallography Open Database(http://www.crystallography.net/cod/)
```cif
_cell_angle_alpha                90.000
_cell_angle_beta                 102.104(8)
_cell_angle_gamma                90.000
_cell_formula_units_Z            4
_cell_length_a                   7.9661(7)
_cell_length_b                   9.205(1)
_cell_length_c                   7.3198(8)
_cell_volume                     524.8(9)
```

[AFlow](https://www.aflowlib.org):
```cif
_cell_length_a  5.4030000000
_cell_length_b  3.4330000000
_cell_length_c  5.0790000000
_cell_angle_alpha  90.0000000000
_cell_angle_beta  132.3200000000
_cell_angle_gamma  90.0000000000
```